test/tablecloth/time/api/slice_test.clj

```diff
@@ -17,76 +15,68 @@
-    _ {:to "1970-01-01T09:00:00.000Z" :from "1970-01-01T10:00:00.000Z"}
-    _ {:to #time/instant "1970-01-01T09:00:00.000Z" :from #time/instant "1970-01-01T10:00:00.000Z"}))
+
+                          (slice (:from arg-map) (:to arg-map))))
+    _ {:from "1970-01-01T09:00:00.000Z" :to "1970-01-01T10:00:00.000Z"}
+    _ {:from #time/instant "1970-01-01T09:00:00.000Z" :to #time/instant "1970-01-01T10:00:00.000Z"}))
 
 (deftest slice-by-local-datetime
   (are [_ arg-map] (= (dataset {:A [#time/date-time "1970-01-01T09:00"
                                     #time/date-time "1970-01-01T10:00"]
                                 :B [9 10]})
                       (-> (dataset {:A (plus-temporal-amount #time/date-time "1970-01-01T00:00" (range 11) :hours)
                                     :B (range 11)})
-                          (index-by :A)
                           (slice (:to arg-map) (:from arg-map))))
     _ {:to "1970-01-01T09:00" :from "1970-01-01T10:00:00"}
     _ {:to #time/date-time "1970-01-01T09:00" :from #time/date-time "1970-01-01T10:00"}))
 
-(deftest slice-by-year
-  (are [_ arg-map] (= (dataset {:A [#time/year "1979" #time/year "1980"]
-                                :B [4 5]})
-                      (-> (dataset {:A [#time/year "1975" #time/year "1976"
-                                        #time/year "1977" #time/year "1978"
-                                        #time/year "1979" #time/year "1980"]
-                                    :B (range 6)})
-                          (index-by :A)
-                          (slice (:to arg-map) (:from arg-map))))
-    _ {:to "1979" :from "1980"}
-    _ {:to #time/year "1979" :from #time/year "1980"}))
-
-;; TODO Fix this. We need to make dtype aware of year-month. This is not done yet.
-;; (deftest slice-by-year-month
-;;   (are [_ arg-map] (= (dataset {:A [#time/year-month "1979-01" #time/year-month "1980-01"]
-;;                                 :B [9 10]})
-;;                       (-> (dataset {:A (plus-temporal-amount #time/year-month "1970-01" (range 11) :years)
-;;                                     :B (range 11)})
-;;                           (index-by :A)
-;;                           (slice (:to arg-map) (:from arg-map))))
-;;     _ {:to "1979-01" :from "1980-01"}
-;;     _ {:to #time/year-month "1979-01" :from #time/year-month "1980-01"}))
-
 (deftest slice-by-local-date
   (are [_ arg-map] (= (dataset {:A [#time/date "1979-01-01" #time/date "1980-01-01"]
                                 :B [9 10]})
                       (-> (dataset {:A (plus-temporal-amount #time/date "1970-01-01" (range 11) :years)
                                     :B (range 11)})
-                          (index-by :A)
                           (slice (:to arg-map) (:from arg-map))))
     _ {:to "1979-01-01" :from "1980-01-01"}
     _ {:to #time/date "1979-01-01" :from #time/date "1980-01-01"}))
 
+(deftest slice-index-with-non-unique-values
```

Comment:
Actually solved in https://github.com/techascent/tech.ml.dataset/pull/214, but we want to ensure support for this here as well I think.